tar/bsdtar.c

```diff
@@ -1399,6 +1414,48 @@ configfile_helper(struct bsdtar *bsdtar, const char *line)
 	return (0);
 }
 
+static void
+archive_names(struct bsdtar *bsdtar, const char *fname)
+{
+	struct stat sb;
+
+	/* If it's a file, does it exist? */
+	if (strcmp(fname, "-") != 0) {
+		if (stat(fname, &sb))
+			bsdtar_errc(bsdtar, 1, errno, "stat(%s)", fname);
+	}
+
+	/* Process the file. */
+	process_lines(bsdtar, fname, archive_names_helper, 0);
+}
+
+/* Process a line of configuration file. */
+static int
+archive_names_helper(struct bsdtar *bsdtar, const char *line)
+{
+	char * name;
+
+	/* Skip any leading whitespace. */
+	while ((line[0] == ' ') || (line[0] == '\t'))
+		line++;
+
+	/* Ignore comments and blank lines. */
+	if ((line[0] == '#') || (line[0] == '\0'))
```

Comment:
... for that matter, someone might have an archive named `#foo`.  I'd be inclined to simply take the archive names as provided.